embassy-usb-dfu/src/application.rs

```diff
@@ -130,8 +130,23 @@ pub fn usb_dfu<'d, D: Driver<'d>, MARK: DfuMarker, RST: Reset>(
     builder: &mut Builder<'d, D>,
     handler: &'d mut Control<MARK, RST>,
     timeout: Duration,
+    winusb_guids: Option<&'d [&str]>,
 ) {
     let mut func = builder.function(0x00, 0x00, 0x00);
+    if let Some(winusb_guids) = winusb_guids {
+        // We add MSOS headers so that the device automatically gets assigned the WinUSB driver on Windows.
+        // Otherwise users need to do this manually using a tool like Zadig.
+        //
+        // Adding them here on the function level appears to only be needed for compositive devices.
+        // In addition to being on the function level, they should also be added to the device level.
+        //
+        func.msos_feature(msos::CompatibleIdFeatureDescriptor::new("WINUSB", ""));
+        func.msos_feature(msos::RegistryPropertyFeatureDescriptor::new(
+            "DeviceInterfaceGUIDs",
+            msos::PropertyData::RegMultiSz(winusb_guids),
+        ));
+    }
+
```

Comment:
Could this have been a closure that gets the func passed, instead of an option parameter that seems very specific to winusb?

Reply:
Great suggestion, thanks! I can't see use-case for customizing `func` further right now, but it's definitely a more future proof approach. I'll give that approach a shot and see if it works out

Reply:
Ok, I have now updated this to use a closure instead. Unfortunately, I could not manage to put the closure inside an `Option` though since that results in all sorts of lifetime hell...